arrays/arrays.c

```diff
@@ -35,25 +38,30 @@ Array *create_array (int capacity) {
 void destroy_array(Array *arr) {
 
   // Free all elements
-
+  free(arr->elements);
   // Free array
-
+  free(arr);
 }
 
 /*****
  * Create a new elements array with double capacity and copy elements
  * from old to new
  *****/
-void resize_array(Array *arr) {
+Array *resize_array(Array *arr) {
 
   // Create a new element storage with double capacity
-
+  Array *newarray = create_array((arr->capacity*2));
```

Comment:
You should be just be creating a new char ** here to replace char **elements. You don't need to create a whole new array struct here.